packages/typespec-client-generator-core/src/lib.ts

```diff
@@ -159,6 +159,12 @@ export const $lib = createTypeSpecLibrary({
         default: `Cannot pass an empty value to the @clientName decorator`,
       },
     },
+    "duplicate-model-name": {
```

Comment:
Is this really just only applicable to models? What about:
- properties
- enum
- union
- interfaces
- operations, etc.
basically everything that clientName affect

Reply:
Good point!
Based on the [Type definition](https://github.com/microsoft/typespec/blob/76bbb4041372f9183be44cd9268d4f4a3c0b34eb/packages/compiler/src/core/types.ts#L112-L136), which can be applied with `clientName`, I summarized the check rule after some discussion with @tadelesh

```
Unique per namespace:
- Namespace
- Interface
- Model & Enum & Union
- Operation

Unique within a specific model:
- ModelProperty

Unique within a specific enum:
- EnumMember

Unique within a Union:
- UnionVariant

```

Reply:
scalar is missing

Reply:
implemented with above rules.